From 1e7c76d702437fb12770318723e892ffe81861f2 Mon Sep 17 00:00:00 2001
Message-Id: <1e7c76d702437fb12770318723e892ffe81861f2.1423288862.git.chang-joon.lee@intel.com>
In-Reply-To: <f73f06e8e99aa68e75d55ffc4fa907c471523e67.1423288862.git.chang-joon.lee@intel.com>
References: <f73f06e8e99aa68e75d55ffc4fa907c471523e67.1423288862.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Sat, 13 Dec 2014 10:12:48 +0530
Subject: [PATCH 3/3] FROM_UPSTREAM [VPG]: drm/i915: Skip gunit save/restore
 for cherryview
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

With cherryview onwards, Gunit hardware itself save and restore all the
Gunit registers. Skipping the "vlv_save_gunit_s0ix_state" &
"vlv_restore_gunit_s0ix_state" for cherryview in S3/S0ix sequence.

Issue: GMINL-4741
Change-Id: I89ae0e633d5630f94bda332f7cbaa5eda6a862c5
Upstream-Repo: git://people.freedesktop.org/~danvet/drm-intel
Upstream-commit: <not available>
Tracked-On: http://lists.freedesktop.org/archives/intel-gfx/2014-December/057340.html
Signed-off-by: Deepak S <deepak.s@intel.com>
Reviewed-by: Ville Syrjälä <ville.syrjala@linux.intel.com>
Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
---
 drivers/gpu/drm/i915/i915_drv.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.c b/drivers/gpu/drm/i915/i915_drv.c
index 1730393..0f8e6d6 100644
--- a/drivers/gpu/drm/i915/i915_drv.c
+++ b/drivers/gpu/drm/i915/i915_drv.c
@@ -1810,7 +1810,9 @@ static int vlv_suspend_complete(struct drm_i915_private *dev_priv)
 	err = vlv_allow_gt_wake(dev_priv, false);
 	if (err)
 		goto err2;
-	vlv_save_gunit_s0ix_state(dev_priv);
+
+	if (!IS_CHERRYVIEW(dev_priv->dev))
+		vlv_save_gunit_s0ix_state(dev_priv);
 
 	err = vlv_force_gfx_clock(dev_priv, false);
 	if (err)
@@ -1848,7 +1850,8 @@ static int vlv_resume_prepare(struct drm_i915_private *dev_priv,
 	if (rpm_resume)
 		ret = vlv_force_gfx_clock(dev_priv, true);
 
-	vlv_restore_gunit_s0ix_state(dev_priv);
+	if (!IS_CHERRYVIEW(dev_priv->dev))
+		vlv_restore_gunit_s0ix_state(dev_priv);
 
 	err = vlv_allow_gt_wake(dev_priv, true);
 	if (!ret)
-- 
1.7.9.5

